javatests/arcs/sdk/android/storage/service/StorageServiceTest.kt

```diff
@@ -0,0 +1,122 @@
+import arcs.android.storage.service.BindingContext
+import arcs.android.storage.service.DeferredResult
+import arcs.android.storage.toParcelable
+import arcs.core.crdt.CrdtCount
+import arcs.core.data.CountType
+import arcs.core.storage.ExistenceCriteria
+import arcs.core.storage.ProxyMessage
+import arcs.core.storage.StorageKey
+import arcs.core.storage.StoreOptions
+import arcs.core.storage.driver.RamDiskStorageKey
+import arcs.sdk.android.storage.ResurrectionHelper
+import com.google.common.truth.Truth.assertThat
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.runBlocking
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.Robolectric
+import org.robolectric.Shadows.shadowOf
+
+@Suppress("EXPERIMENTAL_API_USAGE")
+@RunWith(AndroidJUnit4::class)
+class StorageServiceTest {
+    private lateinit var app: Application
+    private lateinit var storeOptions: StoreOptions<CrdtCount.Data, CrdtCount.Operation, Int>
+
+    @Before
+    fun setUp() {
+        app = ApplicationProvider.getApplicationContext()
+        storeOptions = StoreOptions(
+            RamDiskStorageKey("count"),
+            ExistenceCriteria.MayExist,
+            CountType()
+        )
+    }
+
+    @Test
+    fun sendingProxyMessage_resultsInResurrection() = lifecycle(storeOptions) { service, context ->
+        // Setup:
+        // Create a resurrection helper we'll use to collect updated storage keys coming from the
+        // ShadowApplication-captured nextStartedService intents.
+        val receivedUpdates = mutableListOf<List<StorageKey>>()
+        val resurrectionHelper = ResurrectionHelper(app) { keys: List<StorageKey> ->
+            receivedUpdates.add(keys)
+        }
+
+        // Setup:
+        // Add a resurrection request to the storage service.
+        val resurrectionRequestIntent = Intent(app, StorageService::class.java).apply {
+            ResurrectionRequest.createDefault(app, listOf(storeOptions.storageKey))
+                .populateRequestIntent(this)
+        }
+        service.onStartCommand(resurrectionRequestIntent, 0, 0)
+
+        // Act:
+        // Issue a proxy message to the binding context (and transitively: to the storage service)
+        runBlocking {
+            // Wait to let the resurrection request propagate.
+            while (service.loadJob == null) {
+                delay(100)
+            }
+            service.loadJob?.join()
+
+            val op = CrdtCount.Operation.Increment("foo", 0 to 1)
+            val proxyMessage = ProxyMessage.Operations<CrdtCount.Data, CrdtCount.Operation, Int>(
+                listOf(op), id = 1
+            )
+            val deferredResult = DeferredResult(this.coroutineContext)
+            context.sendProxyMessage(
+                proxyMessage.toParcelable(ParcelableCrdtType.Count),
+                deferredResult
+            )
+
+            deferredResult.await()
+        }.let(::assertThat).isTrue()
```

Comment:
This line is super weird. I can guess at what it does, but surely it'd be better to just use a temporary variable for this?

```
val result = runBlocking { ... }
assertThat(result ?: true).isTrue()
```

Reply:
I was being too clever, yeah.. 🙃